R/checks.R

```diff
@@ -0,0 +1,24 @@
+#' @title Check functions
+#'
+#' @description
+#' These functions are used to check input arguments.
+#'
+#' @param x object to test.
+#'
+#' @return `logical(1)`, `TRUE` if checks are successful otherwise `FALSE`.
+#'
+#' @author Sebastian Gibb
+#' @family helper functions for developers
+#' @export
+#' @examples
+#' isPeaksMatrix(1:2)
+#' isPeaksMatrix(cbind(mz = 1:2, intensity = 1:2))
+isPeaksMatrix <- function(x) {
+    isTRUE(
+        is.matrix(x) &&
+        mode(x) == "numeric" &&
+        ncol(x) == 2L &&
+        !is.null(colnames(x)) &&
+        all(colnames(x) == c("mz", "intensity"))
+    )
```

Comment:
You might also want to check that m/z are ordered increasingly.

Reply:
Indeed. I will add that. In general I am a little bit afraid that all these checks increase our run time dramatically (especially in often called functions like `dotproduct`).

Reply:
Agree. That's why I would **not** call this function within `dotproduct`. But with this function we provide the user a tool to check input *before* calling `dotproduct`.